downloads/swan-lake-release-notes/swan-lake-2201.9.0/RELEASE_NOTE.md

```diff
@@ -461,7 +461,29 @@ $= string value = myClient->/root/name("input");
 $= future<int> result = start name();
 ```
 
-#### CLI
+#### Test Framework
+
+- The test framework now supports the parallel execution of tests as an experimental feature. To activate the parallel test execution, simply use the `--parallel` flag with the `bal test` command.
+
+    ```ballerina
+    bal test --parallel
+    ```
+
+- APIs for mocking client resources is introduced. With these, a client resource can be stubbed to behave in a certain way, previously limited to test double. 
+
+    ```ballerina
+    // Sample HTTP Client call
+
+    http:Client jokes = check new ("https://api.chucknorris.io/jokes/");
+    json joke = check jokes->/random;
+
+    // Stub to return a specific value
+    jokes = test:mock(http:Client);
+    test:prepare(jokes)
+        .whenResource("::path")
+        .withPathParameters({path: ["random"]})
```

Comment:
Can you elaborate on this?  We can have scenarios like below
```ballerina
test:prepare(clientEndpoint).whenResource("greeting/hello2/:name/:town")
    .withPathParameters({town: "London", name: "John"})
    .withArguments("bar", "str").onMethod("post").thenReturn("Hello, John");
```

Reply:
Adding to what Asma has mentioned,
In the example in the release note `path` is just a placeholder. It works the same for any string placeholder.

Reply:
Sorry, what I meant to say was one rest param (therefore, only one with :: syntax). But since values are provided with `withPathParameters` which applies to both rest and non-rest stuff, I guess we need the name.